src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1221,6 +1225,68 @@ private void VisitStatements(IEnumerable<IOperation> statements)
             }
         }
 
+        public override IOperation VisitConstructorBodyOperation(IConstructorBodyOperation operation, int? captureIdForResult)
+        {
+            Debug.Assert(_currentStatement == operation);
+
+            bool haveLocals = !operation.Locals.IsEmpty;
+            if (haveLocals)
+            {
+                EnterRegion(new RegionBuilder(ControlFlowGraph.RegionKind.Locals, locals: operation.Locals));
+            }
+
+            if (operation.Initializer != null)
+            {
+                VisitStatement(operation.Initializer);
+            }
+
+            if (haveLocals)
```

Comment:
This is not correct. Locals declared via declaration patterns in a constructor initializer _are_ visible in the body.

Reply:
For example, this is completely valid code (under C# 7.3):
```C#
class C : Base
{
    C() : base(out var i)
    {
        O(i);
    }
    void O(int i)
    {
        i = 1;
    }
}

class Base
{
    protected Base(out int i)
    {
        i = 1;
    }
}
```

---
In reply to: [189345445](https://github.com/dotnet/roslyn/pull/26955#discussion_r189345445) [](ancestors = 189345445)

Reply:
Ah, I got confused by the doc comment: http://source.roslyn.io/#Microsoft.CodeAnalysis/Operations/IConstructorBodyOperation%2520.cs,21. Let me adjust the implementation and test baselines.

Reply:
Please also add a test like the one above, that way cfg verifier will fail if in the future we break this.

---
In reply to: [189346487](https://github.com/dotnet/roslyn/pull/26955#discussion_r189346487) [](ancestors = 189346487)

Reply:
@333fred I can add a test, but note that existing test `ConstructorBody_14` does guard this case. The fact that the declared local is used or not in the body is not relevant.

Reply:
Yes it is. @AlekseyTs ensured that the CFG Verifier will verify that all locals used are contained in the surrounding scope, and it would have failed the above test case.

---
In reply to: [189379223](https://github.com/dotnet/roslyn/pull/26955#discussion_r189379223) [](ancestors = 189379223)